src/org/labkey/test/pages/core/login/LoginConfigurePage.java

```diff
@@ -20,61 +22,64 @@ public LoginConfigurePage(WebDriver driver)
+                clickSubMenu(false, authenticationProvider.getProviderName() + " : " + authenticationProvider.getProviderDescription());
 
-        elementCache().addMenu.
-                clickSubMenu(true, page.getProviderName() + " - " + page.getProviderDescription());
+        return authenticationProvider.getNewDialog(getDriver());
+    }
 
-        return page;
+    public LoginConfigurePage togglePrimaryConfiguration()
+    {
+        if (!elementCache().panelTab1.getAttribute("aria-selected").equals("true"))
+            elementCache().panelTab1.click();
+        waitFor(()-> elementCache().panelTab1.getAttribute("aria-selected").equals("true"), 1000);
+        return this;
     }
 
-    public LoginConfigurePage removeConfiguration(String description)
+    public LoginConfigRow getPrimaryConfigurationRow(String description)
     {
-        // the html table class does not support finding a row's index correctly; doing it ourself here
-        WebElement row = Locator.xpath("//tbody/tr").containing(description).findElement(elementCache().tableElement);
-        Locator.tagWithClass("a", "labkey-text-link").withText("delete").findElement(row).click();
-        // handle dialog
-        acceptAlert();
-        // pause, because that works
-        sleep(1000);
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver()).withDescription(description).waitFor();
+    }
+
+    public List<LoginConfigRow> getConfigurations()
+    {
+        togglePrimaryConfiguration();
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver()).findAll();
+    }
+
+    public LoginConfigurePage toggleSecondaryConfiguration()
+    {
+        if (!elementCache().panelTab2.getAttribute("aria-selected").equals("true"))
+            elementCache().panelTab2.click();
+        waitFor(()-> elementCache().panelTab2.getAttribute("aria-selected").equals("true"), 1000);
         return this;
     }
 
-    public <P extends PrimaryAuthenticationProviderConfigurationPage> P clickEditConfiguration(Class<P> authType, String description)
+    public LoginConfigurePage removeConfiguration(String description)      // assumes for now we're doing primary only
     {
-        P page;
-        try
-        {
-            page = authType.getDeclaredConstructor(WebDriver.class).newInstance(getDriver());
-        }
-        catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e)
-        {
-            throw new RuntimeException("Unable to instantiate page class: " + authType.getName(), e);
-        }
+        new LoginConfigRow.LoginConfigRowFinder(getDriver()).withDescription(description).waitFor()
+                .clickDelete();
 
-        WebElement row = Locator.xpath("//tbody/tr").containing(description).findElement(elementCache().tableElement);
-        clickAndWait(Locator.tagWithClass("a", "labkey-text-link").withText("edit").findElement(row));
+        // pause, because that works?
+        sleep(1000);
```

Comment:
`LoginConfigRow.clickDelete` should handle this timing.

Reply:
It just happens that the timing in clickDelete has changed; deleting a configuration now puts up a prompt (which clickDelete handles, then waits for the row to become stale)
...so this sleep (a holdover from the old UI) was probably unnecessary